### PR TITLE
Reconnect to sut after reboot on XEN

### PR DESCRIPTION
**Pull request by kravciak**
Fixes:
https://openqa.suse.de/tests/806964#step/transactional_update/22 XEN hvm
https://openqa.suse.de/tests/806966#step/transactional_update/22 XEN pv

Requires: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/324

Local runs:
http://dhcp91.suse.cz/tests/4107 - XEN pv
http://dhcp91.suse.cz/tests/4117 - XEN hvm
http://dhcp91.suse.cz/tests/4103 - KVM
http://dhcp91.suse.cz/tests/4105 - DVD

### CODE REVIEW

File: lib/caasp.pm [hunk 1]
```diff
@@ -33,6 +33,11 @@ sub process_reboot {
     script_run("reboot", 0) if $trigger;
 
     reset_consoles;
+    if (check_var('VIRSH_VMM_FAMILY', 'xen')) {
+        sleep 7;
+        select_console 'sut';
+    }
```
Comment (okurz) [inline lib/caasp.pm]: it does not look correct to me to have something like this in `lib/caasp.pm`. That is valid regardless of the product but depending on the backend, right?

Comment (kravciak) [inline lib/caasp.pm]: Well, it's not implemented in wait_boot and every test module handles this itself for now
 # grep -ri "select_console.*sut" found 7 examples

Comment (kravciak) [inline lib/caasp.pm]: I am looking at option to put it into wait_boot

Comment (kravciak) [inline lib/caasp.pm]: @mnowaksuse Will I break something if I move it here? https://github.com/kravciak/openqa-tests/blob/casp-xen1/lib/opensusebasetest.pm#L218

What about timeout? - sometimes it takes longer to reboot, might be a reason to keep per-module solution. I don't feel great about moving it to wait_boot - what do you think?

Comment (kravciak) [inline lib/caasp.pm]: I discussed with Michal, he will test another (more systematic) solution.

Comment (mnowaksuse) [inline lib/caasp.pm]: I'd rather not use `wait_boot` for non-SLES/openSUSE distros at all, as it's really not needed and only adds complexity. And sure we need a 'reconnect' solution in os-autoinst; will file ticket about that.

**Comment by kravciak** [comment]
Discussed with Michal, we think putting "if CaaSP & XEN then reconnect & return" into wait_boot will clutter it even more. I agree that reconnecting in every module that reboots is not systematic solution.

@sysrich is working on grub output to serial console, I propose to wait until it's done and change XEN reconnecting then. WDYT Oliver?

**Comment by kravciak** [comment]
poo#17658

**Comment by kravciak** [comment]
Local runs:
dhcp91.suse.cz/tests/4107 - XEN
dhcp91.suse.cz/tests/4103 - KVM
dhcp91.suse.cz/tests/4105 - DVD

**Comment by okurz** [comment]
sure, good enough for me

**Comment by sysrich** [comment]
> @sysrich is working on grub output to serial console, I propose to wait until it's done and change XEN reconnecting then

I am? says who?

**Pull request merged**